src/anndata/tests/helpers.py

```diff
@@ -331,8 +331,8 @@ def gen_adata(  # noqa: PLR0913
         random_state = np.random.default_rng()
 
     M, N = shape
-    obs_names = pd.Index(f"cell{i}" for i in range(shape[0]))
-    var_names = pd.Index(f"gene{i}" for i in range(shape[1]))
+    obs_names = pd.Index([f"cell{i}" for i in range(shape[0])], dtype=str)
+    var_names = pd.Index([f"gene{i}" for i in range(shape[1])], dtype=str)
```

Comment:
this is why I add `lambda:` in a lot of places: `gen_adata` needs to be called when the `infer_strings` option is as we want it.

Reply:
This is a great thing to fix anyway.  This isn't the only case we'll want `gen_adata` run under various conditions